### DISCUSSION TITLE
Tag customization and recovery

### DISCUSSION THREAD

**Discussion by esdcharles** [Azure/ipam#105] [Q&A]
Hi there,

Apologies in advance if I missed this in the docs or here in the discussions.  

I had a couple questions around the tags involved with the reservations:

1) Some management have requested if there is a way to customize tags in a "...more friendly manner?"  For example, when a customer makes a reservation we can customize it something our group would prefer as opposed to X-IPAM-RES-ID-xxxxx. Not sure if this a tough ask or not but thought I'd put it out there.

2) If a customer "misplaces" the tag they were issued is there a way of locating it and if not, is there a method/process to reissue the tag to be used with their VNet. Perhaps I have the role of the tag misunderstood  - if so please let me know as I am in the very early stages of testing with IPAM.

Thanks!

**Comment by DCMattyG** [comment]
Hello there @binarybfc, thanks you so much for reaching out today! Responses as follows...

1. I'm trying to see how having unique Tag identifiers will help...can you spell out a bit better for me why a standardized Tag approach wouldn't work well for your specific environment? I'm genuinely curious, and happy to make adjustments for any request that makes sense. The *value* of that Tag is uniquely mapped back to the **Space/Block** so from and Azure IPAM perspective, it wouldn't be mixing up anything.
2. Yes, you can find the ID again by either querying the API to list the **Reservations** for a given **Block** (```/api/spaces/{space}/blocks/{blocks}/reservations```), or in the UI you can navigate to **Configuration -> Space -> Block -> Reservations** and view them there. The UI even allows you to copy them directly to the clipboard.

I hope that helps, but we're always here for additional clarity, guidance, and assistance!

**Comment by esdcharles** [comment]
Thanks for the reply. I can completely understand the curiosity. The ask from my group was really about having the choice of naming scheme for the IDs being attached . We still need to have the Unique IDs and understand the reasoning behind it but would it be nice to have that option to customize them like below:
![image](https://user-images.githubusercontent.com/102684496/229186993-5d791992-d814-4960-9a53-ea071709106d.png)

Thanks so much once again!

**Reply by DCMattyG** [reply]
Of course, thanks for the details!

One last clarification....would the ask be to change this value so all users would use "X-IPAM-RES-ID", or in the case above "IPAMReservationID" per IPAM instance? Or would you like to have multiple different possible TAG values/possibilities per IPAM instance?

**Reply by esdcharles** [reply]
Honestly I am good with just being able to have the option to change the tag value i.e. IPAMReservationID (or any other value we opt to go with) attached to each IPAM instance. Thanks!

PS While on the topic of tags I also caught this from @XtratusCloud and really liked his idea- this would also come in handy as well 

https://github.com/Azure/ipam/issues/84

**Reply by DCMattyG** [reply]
Understood, and the item from #84 has already been incorporated into my latest branch 👍
